### PR TITLE
Allow to dump signer certificate without verification

### PR DESCRIPTION
**Pull request by ejoerns**
Especially for debugging-purpose, it is quite useful to allow dumping the signer certificate also without having to verify it before.

### CODE REVIEW

File: test/rauc.t [hunk 1]
```diff
@@ -202,6 +202,11 @@ test_expect_success "rauc info" "
     info $SHARNESS_TEST_DIRECTORY/good-bundle.raucb
 "
 
+test_expect_success "rauc info dump-cert unverified" "
+  rauc --no-verify --dump-cert \
+    info $SHARNESS_TEST_DIRECTORY/good-bundle.raucb
```
Comment (jluebbe) [inline test/rauc.t]: Shouldn't we test this on bad bundle?

File: src/signature.c [hunk 2]
```diff
@@ -522,6 +522,41 @@ gchar* print_signer_cert(STACK_OF(X509) *verified_chain)
...
+	if (signers == NULL) {
+		g_set_error_literal(
+				error,
+				R_SIGNATURE_ERROR,
+				R_SIGNATURE_ERROR_GET_SIGNER,
+				"Failed to obtain signer info");
+		return NULL;
+	}
+
+	ret = dump_cms(signers);
```
Comment (jluebbe) [inline src/signature.c]: Should you free the signers here?

File: src/signature.c [hunk 3]
```diff
@@ -522,6 +522,41 @@ gchar* print_signer_cert(STACK_OF(X509) *verified_chain)
 	return ret;
 }
 
+gchar* print_signer_cert(GBytes *sig, GError **error)
```
Comment (jluebbe) [inline src/signature.c]: This doesn't print the cert but returns it as a human-readable string.

File: src/signature.c [hunk 4]
```diff
@@ -502,7 +502,7 @@ gchar** get_pubkey_hashes(STACK_OF(X509) *verified_chain)
 	return ret;
 }
 
-gchar* print_signer_cert(STACK_OF(X509) *verified_chain)
+static gchar* dump_cms(STACK_OF(X509) *verified_chain)
```
Comment (jluebbe) [inline src/signature.c]: The verified_chain argument could use a better name.

File: include/signature.h [hunk 5]
```diff
@@ -150,4 +151,6 @@ gchar* print_cert_chain(STACK_OF(X509) *verified_chain);
  *
  * @return TRUE if succeeded, FALSE if failed
  */
-gboolean cms_get_cert_chain(CMS_ContentInfo *cms, X509_STORE *store, STACK_OF(X509) **verified_chain, GError **error);;
+gboolean cms_get_cert_chain(CMS_ContentInfo *cms, X509_STORE *store, STACK_OF(X509) **verified_chain, GError **error);
+
+gboolean cms_info_file(const gchar *filename, GBytes *sig, gsize limit, CMS_ContentInfo **cms, GError **error);
```
Comment (jluebbe) [inline include/signature.h]: This doesn't match any function.

File: test/rauc.t (hunk 1 above)
Comment (ejoerns) [reply test/rauc.t]: The relevant check for this patch is that --dump-cert works without verification. That verification is note performed here is assured as we do not provide a keyring.

However, despite a bit unrelated, I now added a check to assure that rauc info fails verification upon invalid signature.

**Comment by jluebbe** [comment]
Looks good to me now. I've rebased onto master and squashed the fixup.

**Pull request merged**